src/whenever/__init__.py

```diff
@@ -607,11 +612,14 @@ def from_canonical_format(cls, s: str, /) -> Time:
 
         Raises
         ------
-        InvalidFormat
+        ValueError
             If the string does not match this exact format.
         """
         if not _match_time(s):
-            raise InvalidFormat()
+            raise ValueError(
+                "Could not parse as canonical format "
```

Comment:
One message for two code paths (canonical and common ISO 8601) is too confusing. Let's just bite the bullet and have some code duplication—or, some reusable utility function.

Reply:
Let's also make this change in other methods that have two code paths

Reply:
Done, with utility functions. Adding them helped me address the sticking to a single generic error without chained detailed errors, see below.